source/server/options_impl.cc

```diff
@@ -19,7 +19,7 @@
 
 namespace Envoy {
 namespace {
-std::vector<std::string> toArgsVector(int argc, const char* const* argv) {
+std::vector<std::string> toArgsVector(std::size_t argc, const char* const* argv) {
```

Comment:
Can you clarify, how this breaks windows, please? How is std::array class gets involved here.

Reply:
Happy to clarify. Please note, this is half the patch (which we assembled from the wrong working tree) and will be further corrected in the morning.

On windows, size_t is the size of a pointer (64 bits), whereas an int is 32 bits.

On linux, size_t is the size of a pointer which is the size of an int which is 64 bits, which is why this goes unnoticed on that architecture.

The present code is unworkable, as noted by many other instances in the options_impl test suite sources by correctly using size_t as the loop indexes (and we missed one which is why the tests currently fail on gcc.) With the second commit to be submitted, the tests and the core implementation are correct, irrespective of platform, based on the underlying std::array declaration according to spec.

Reply:
An alternate fix is to recast all std::array param results by casting theoptions.size() to an int before passing this to OptionsImpl's creator. Seems sub-optimal.

Reply:
The other half of this patch is applied, and passing all tests in CI and locally here on Windows.

Reply:
I assume there is some other change in some other PR where this causes build problems, right? I can't imagine how this code would fail to compile on its own. It is after all there to support parsing `main` function parameters and as such int was used for argc.

Reply:
@yanavlasov see this part of the source file: https://github.com/envoyproxy/envoy/blob/8fcb780171c960d3adfce33324b6e304da567d0d/source/server/options_impl.cc#L34

Reply:
Note that argc in this context is describing the number of elts, 
https://en.cppreference.com/w/cpp/container/array

and bears no relationship to the traditional argc of the main() function, which is declared altogether differently in the spec;
https://en.cppreference.com/w/cpp/language/main_function

Reply:
@yanavlasov is this now clear, or are you looking for further explanation?

Reply:
Reviewing, it's apparent that this is tied to main()'s definition of argc, went ahead as suggested to conform to that declaration.